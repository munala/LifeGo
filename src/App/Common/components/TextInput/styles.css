.container-input {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: stretch;
  flex-grow: 1;
  align-self: center;
}
