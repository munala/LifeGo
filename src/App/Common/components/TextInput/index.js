import React from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import { withStyles } from '@material-ui/core/styles';

import './styles.css';

const styles = theme => ({
  textField: {
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit,
    width: 200,
  },
});

const TextInput = ({
  autoFocus,
  classes,
  style,
  value,
  onChange,
  rowsMax,
  placeholder,
  disabled,
  error,
  disableUnderline,
  Children,
}) => (
  <div className="container" >
    <TextField
      className={classes.textField}
      autoFocus={autoFocus}
      placeholder={placeholder}
      color="primary"
      value={value}
      rowsMax={rowsMax}
      onChange={onChange}
      disabled={disabled}
      error={error}
      disableUnderline={disableUnderline}
      style={style}
    >
      {Children && <Children />}
    </TextField>
  </div>
);

TextInput.propTypes = {
  value: PropTypes.string.isRequired,
  autoFocus: PropTypes.bool,
  placeholder: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  style: PropTypes.shape({}),
  rowsMax: PropTypes.number,
  classes: PropTypes.shape({}).isRequired,
  error: PropTypes.bool,
  disableUnderline: PropTypes.bool,
  Children: PropTypes.element,
};

TextInput.defaultProps = {
  placeholder: undefined,
  autoFocus: false,
  disabled: false,
  style: undefined,
  rowsMax: undefined,
  Children: undefined,
  error: false,
  disableUnderline: false,
};

export default withStyles(styles)(TextInput);
