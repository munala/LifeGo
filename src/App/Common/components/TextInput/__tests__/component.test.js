import React from 'react';
import { shallow } from 'enzyme';
import TextInput from '../';

describe('TextInput tests', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<TextInput
      name="name"
      value="value"
      onChange={() => {}}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
