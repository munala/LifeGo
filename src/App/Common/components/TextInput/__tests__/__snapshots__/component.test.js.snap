// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput tests renders without crashing 1`] = `
<TextInput
  autoFocus={false}
  classes={
    Object {
      "menu": "TextInput-menu-2",
      "textField": "TextInput-textField-1",
    }
  }
  disableUnderline={false}
  disabled={false}
  error={false}
  multiline={false}
  name="name"
  onChange={[Function]}
  onKeyPress={[Function]}
  value="value"
/>
`;
